src/components/Header.svelte

```diff
@@ -1,34 +1,136 @@
 <script>
   import logo from '/src/assets/svelte.svg';
-  import { Button } from 'flowbite-svelte';
-  import { link, location } from 'svelte-spa-router';
+  import { link, location, push } from 'svelte-spa-router';
+  import { Button, Input } from 'flowbite-svelte';
+  import placesService from '../services/PlacesService.js';
+  import { clickOutside } from '../directives/clickOutside.js';
+  import { debounce } from 'lodash';
 
-  const dummyCity = 'Ukraine, Kyiv, elevation 199m';
-  // TODO replace dummyCity once favorite places ready
+  let searchInputContainerRef;
 
+  let searchValue = '';
+  let placesAutocomplete = [];
+  let isSearchButton = true;
+  let isAutocompleteLoading = false;
+  const dummyCity = 'Ukraine, Kyiv, elevation 199m'; // TODO replace dummyCity once favorite places ready
+
+  $: searchInputRef = searchInputContainerRef?.querySelector('input');
+
+  $: isNoData = !placesAutocomplete.length && searchValue && !isAutocompleteLoading;
   $: isSearchPage = $location.includes('/search');
+
+  $: {
+    if (!searchValue) {
+      placesAutocomplete = [];
+      searchInputRef?.focus();
+    } else {
+      getAutocomplete(searchValue);
+    }
+  }
+
+  const toggleIsSearchButton = () => {
+    searchValue = '';
+    isSearchButton = !isSearchButton;
+  };
+
+  const updateSearchValue = debounce(async value => {
+    searchValue = value;
+  }, 300);
+
+  const getAutocomplete = async value => {
+    isAutocompleteLoading = true;
+
+    const { predictions } = await placesService.getAutocomplete(value);
+
+    placesAutocomplete = [...predictions];
+    isAutocompleteLoading = false;
+  };
+
+  const navigateToForecast = async placeId => {
+    const { lat, lng } = (await placesService.getPlace(placeId)).result.geometry.location;
+
+    toggleIsSearchButton();
+    push(`/forecast/${lat}/${lng}`);
+  };
+
+  const onKeydown = e => {
+    if (e.key === 'Enter') {
+      navigateToForecast(placesAutocomplete[0]['place_id']);
```

Comment:
I'm not sure. Do you think it should be navigate to the first place from the autocomplete? It seems to me that in this case we can redirect to the search page and display all results from autocomplete. Additionally, when we insert `space` and press `enter` it'll throw the error. Maybe when user input something invalid we can check it and also redirect to the search page with some message about unsuccessful match.

Reply:
About navigation to the first place from the autocomplete it's done like in google 
![image](https://github.com/user-attachments/assets/a02db1a5-97ff-456f-86df-e19ef764177d)
About error with space and press enter good point, I'll add `trim() `to searchValue 👌